Integrate content from multitenancy.md and service_path.md (#4192)
telefonicaid/fiware-orion: doc/manuals/orion-api.md

you were implementing client-side expiration based on the value of that attribute, as your entities could
be automatically deleted in an unwanted way**.

## Multi tenancy

> Orion Context Broker implements a simple multitenant/multiservice

Reviewer: ```suggestion
Orion implements a simple multitenant/multiservice
```

> The Orion Context Broker implements a simple multitenant/multiservice
> model based and logical database separation, to ease service/tenant
> based authorization policies provided by other FIWARE components or
> third party software, e.g. the ones in the FIWARE security framework
> (PEP proxy, IDM and Access Control). This functionality is activated
> when the `-multiservice` command line option is used. When
> `-multiservice` is used, Orion uses the `Fiware-Service` HTTP header in
> the request to identify the service/tenant. If the header is not present
> in the HTTP request, the default service/tenant is used.

Reviewer: ```suggestion
Orion Context Broker implements a simple multitenant/multiservice
model based and logical database separation, to ease service/tenant
based authorization policies provided by other FIWARE components or
third party software, e.g. the ones in the FIWARE security framework
(PEP proxy, IDM and Access Control). This functionality is activated
when the `-multiservice` [command line option](admin/cli.md) is used. When
`-multiservice` is used, Orion uses the `Fiware-Service` HTTP header in
the request to identify the service/tenant. If the header is not present
in the HTTP request, the default service/tenant is used.
```

> Multitenant/multiservice ensures that the
> entities/attributes/subscriptions of one service/tenant are *"invisible"*
> to other services/tentants. For example, queryContext on tenantA space
> will never return entities/attributes from tenantB space. This isolation
> is based on database separation, which [details are described in the
> Installation and Administration
> manual](admin/database_admin.md#multiservicemultitenant-database-separation).

Reviewer: ```suggestion
Multitenant/multiservice ensures that the
entities/attributes/subscriptions of one service/tenant are *"invisible"*
to other services/tentants. For example, `GET /v2/entities` on tenantA space
will never return entities from tenantB space. This isolation
is based on database separation, which [details are described in the
Installation and Administration
manual](admin/database_admin.md#multiservicemultitenant-database-separation).
```

In addition, note that when `-multiservice` is used Orion includes the
`Fiware-Service` header in the notifyContextRequest request messages associated to subscriptions
in the given tenant/service (except for the default service/tenant, in
which case the header is not present), e.g.:

```
POST http://127.0.0.1:9977/notify
Content-Length: 725
User-Agent: orion/0.13.0
Host: 127.0.0.1:9977
Accept: application/json
Fiware-Service: t_02
Content-Type: application/json

{
...
}
```

> Regarding service/tenant name syntax, it must be a string of
> alphanumeric characters (and the `\` symbol). Maximum length is 50
> characters,
> which should be enough for most use cases. Orion Context Broker
> interprets the tenant name in lowercase, thus, although you can use
> tenants such as in updateContext `MyService` it is not advisable, as the
> notifications related with that tenant will be sent with `myservice`
> and, in that sense, it is not coherent the tenant you used in
> updateContext compared with the one that Orion sends in
> notifyContextRequest.

Reviewer: ```suggestion
Regarding service/tenant name syntax, it must be a string of
alphanumeric characters (and the `\` symbol). Maximum length is 50
characters,
which should be enough for most use cases. Orion Context Broker
interprets the tenant name in lowercase, thus, although you can use
tenants such as in update `MyService` it is not advisable, as the
notifications related with that tenant will be sent with `myservice`
and, in that sense, it is not coherent the tenant you used in
update requests compared with the one that Orion sends in
notifications.
```

## Service paths

### Entity service paths

> Orion Context Broker supports hierarchical scopes, so entities can be

Reviewer: ```suggestion
Orion supports hierarchical scopes, so entities can be
```

    different Scopes. E.g. we can create entity ID `Tree1` of type
    `Tree` in `/Madrid/Gardens/ParqueNorte/Parterre1` and another entity
    with ID `Tree1` of type `Tree` in `Madrid/Gardens/ParqueOeste` without
    getting any error. However, queryContext can be weird in this
    scenario (e.g. a queryContext in `Fiware-ServicePath /Madrid/Gardens`
    will returns two entities with the same ID and type in the
    queryContextResponse, making hard to distinguish to which scope
    belongs each one)

-   Entities belongs to one (and only one) scope.

-   `Fiware-ServicePath` header is included in notification requests sent by Orion.

-   The scopes entities can be combined orthogonally with the
    [multi-tenancy functionality](#multi-tenancy). In that case,
    each `scope tree` lives in a different service/tenant and they can
    use even the same names with complete database-based isolation. See
    figure below.

![](ServicePathWithMultiservice.png "ServicePathWithMultiservice.png")

-   Current version doesn’t allow to change the scope to which an entity
    belongs through the API (a workaround is to modify the

>     `id.servicePath` field in the entities collection directly).

Reviewer: ```suggestion
    `_id.servicePath` field in the [entities collection](admin/database_model.md#entities-collection) directly).
```
